modules/libcom/RTEMS/epicsNtp.c

```diff
@@ -0,0 +1,86 @@
+/*************************************************************************\
+* (C) 2014 David Lettier.
+* http://www.lettier.com/
+* EPICS BASE is distributed subject to a Software License Agreement found
+* in file LICENSE that is included with this distribution.
+\*************************************************************************/
+
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+#include <rtems.h>
+#include "epicsNtp.h"
+
+int epicsNtpGetTime(char *ntpIp, struct timespec *now)
```

Comment:
I don't see where this routine is getting called, are these files actually needed at all?

Reply:
It seems like its being called from modules/libcom/RTEMS/posix/rtems_init.c:
```
1027    if (epicsNtpGetTime(rtemsInit_NTP_server_ip, &now) < 0) {                                                                                                                        
1028      printf ("***** Can't get time from ntp ...\n");                                                                                                                                
1029    } else {                                                                                                                                                                         
1030      if (clock_settime(CLOCK_REALTIME, &now) < 0){                                                                                                                                  
1031        printf ("***** Can't set time: %s\n", rtems_status_text (sc));                                                                                                               
1032      } else {                                                                                                                                                                       
1033        strftime(timeBuff, sizeof timeBuff, "%D %T", gmtime(&now.tv_sec));                                                                                                           
1034        printf("time from ntp : %s.%09ld UTC\n", timeBuff, now.tv_nsec);   
```

Reply:
Ah, and GitHub wasn't showing the contents of that new file so my browser search didn't find it – my bad!